es6-shim.js

```diff
@@ -2009,6 +2016,9 @@
         capability.resolve = resolve;
         capability.reject = reject;
       };
+      // Initialize fields to inform optimizers about the object shape.
```

Comment:
That doesn't seem to be a problem when I look at the generated optimized assembly.

And it's a little nontrivial, since the spec explicitly says the type should be undefined, that is, it allows you to call `resolver(undefined, undefined)` before actually passing the real resolver functions and it should be a no-op.  So the type check in line 2006/2013 above would have to be something like:
`if ((capability.resolve !== void 0 && capability.resolve !== NOOP_FUNC) || ...`
and the extra complexity isn't worth it -- again, since looking at the optimized assembly seems to indicate it doesn't matter.